pd/src/main.rs

```diff
@@ -114,18 +132,35 @@ enum TestnetCommand {
 
     /// Like `testnet generate`, but joins the testnet to which the specified node belongs
     Join {
-        #[clap(default_value = "testnet.penumbra.zone")]
-        node: String,
-        // Default: node-#
+        /// URL of the remote Tendermint RPC endpoint for bootstrapping connection.
+        #[clap(
+            env = "PENUMBRA_PD_JOIN_URL",
+            default_value = "http://testnet.penumbra.zone:26657"
+        )]
+        node: Url,
         /// Human-readable name to identify node on network
         // Default: 'node-#'
-        #[clap(long)]
+        #[clap(long, env = "PENUMBRA_PD_TM_MONIKER")]
         moniker: Option<String>,
-        /// Public IP address to advertise for this node's Tendermint P2P service.
+        /// Public URL to advertise for this node's Tendermint P2P service.
         /// Setting this option will instruct other nodes on the network to connect
-        /// to yours.
-        #[clap(long)]
-        external_address: Option<String>,
+        /// to yours. Must be in the form of a full URL, e.g. "tcp://1.2.3.4:26656".
+        #[clap(long, env = "PENUMBRA_PD_TM_EXTERNAL_URL")]
+        external_address: Option<Url>,
+        /// When generating Tendermint config, use this socket to bind the Tendermint RPC service.
+        #[clap(
+            long,
+            env = "PENUMBRA_PD_TM_RPC_BIND_ADDR",
+            default_value = "0.0.0.0:26657"
+        )]
+        tendermint_rpc_bind_addr: SocketAddr,
+        /// When generating Tendermint config, use this socket to bind the Tendermint P2P service.
+        #[clap(
+            long,
+            env = "PENUMBRA_PD_TM_P2P_BIND_ADDR",
+            default_value = "0.0.0.0:26656"
+        )]
+        tendermint_p2p_bind_addr: SocketAddr,
```

Comment:
This may not matter, but one thing I noticed looking at the `pd start` options is that there was a consistent terminology between `foo_bind` (specifying inbound addresses) and `foo_addr` (specifying outbound addresses), so that you can tell just by glancing at the name whether a parameter is an inbound or outbound address.  Should we align these with that, and make them `tendermint_foo_bind` rather than `tendermint_foo_bind_addr`?